### PR TITLE
Rest Client Reactive - Jackson extension

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

**Comment by geoand** [comment]
I unfortunately don't have time to review this today - Quarkus World Tour is calling :).

@FroMage maybe you can take a look?

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>quarkus-rest-client-reactive-jackson-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <artifactId>quarkus-rest-client-reactive-jackson</artifactId>
+    <name>Quarkus - MicroProfile Rest Client with Reactive RestEasy and Jackson</name>
+    <description>Jackson integration for RESTEasy Reactive MicroProfile Rest Client.</description>
```
Comment (gsmet) [inline extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml:13]: Very minor but we don't have dots for the others (and this appears at code.quarkus.io)

**Review by gsmet — DISMISSED** [review]
Added a very minor micro comment. Please dismiss when fixed :).

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson-common/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProviderDefinedBuildItem.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package io.quarkus.resteasy.reactive.jackson.deployment.processor;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public class ResteasyReactiveJacksonProviderDefinedBuildItem extends MultiBuildItem {
```
Comment (FroMage) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson-common/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProviderDefinedBuildItem.java:5]: I think we should always add javadoc to our build items.

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java [hunk 3]
```diff
@@ -47,7 +46,6 @@ public Person getPerson() {
         return person;
     }
 
-    @CustomSerialization(UnquotedFieldsPersonBiFunction.class)
```
Comment (FroMage) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java:50]: Why is this moved out? I don't see any test replacing this and it's confusing if the test still passes, no?

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/resources/quarkus-extension.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+name: "MicroProfile Rest Client with RESTEasy Reactive and Jackson"
```
Comment (FroMage) [inline extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/resources/quarkus-extension.yaml:2]: The equivalent `rest-client-jackson` has `REST Client Jackson` for name, whereas this one seems to tell me this extension is a full rest client, but it's not, it's support for jackson for the rest client:

```
name: "REST Client Jackson"
metadata:
  keywords:
  - "rest-client-jackson"
  - "rest-client"
  - "web-client"
  - "microprofile-rest-client"
  - "json"
  - "jackson"
  categories:
  - "web"
  - "serialization"
  status: "stable"
```

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml (hunk 1 above)
Comment (FroMage) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml:13]: I guess the `quarkus-rest-client-jackson` equivalent is `Enable Jackson serialization for the REST Client`, so we should harmonise those two

**Review by FroMage — CHANGES_REQUESTED** [review]
LGTM except some cosmetics :)

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml (hunk 1 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml:13]: the description is based on https://github.com/quarkusio/quarkus/blob/a1f4655416bac5b59a157b5840ce52c8c9d48015/extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/pom.xml#L14 which also has a comma at the end :)

I can change both descriptions to follow RestEasy classic. WDYT?

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java (hunk 3 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java:50]: It was not intended.

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/resources/quarkus-extension.yaml (hunk 4 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/resources/quarkus-extension.yaml:2]: I'm not sure I follow. Both extensions, quarkus-rest-client-jackson and quarkus-rest-client-reactive-jackson bring in the client extension. It's enough to add the `-jackson` extension to the project to use the client. 
So for the content of the extension point of view, indeed it just adds jackson to the client. But for users, adding the extension adds the client with jackson support to their project.

OTOH, the current name is rather long :)

Comment (FroMage) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/resources/quarkus-extension.yaml:2]: I guess I agree with you, but this is not how it's described for RESTEasy Classic. We have to harmonise both.

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml (hunk 1 above)
Comment (FroMage) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml:13]: Yes, they have to be harmonised, ideally.

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson-common/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProviderDefinedBuildItem.java (hunk 2 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson-common/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProviderDefinedBuildItem.java:5]: done

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java (hunk 3 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java:50]: fixed

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/resources/quarkus-extension.yaml (hunk 4 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/resources/quarkus-extension.yaml]: I reworked it to align names and descriptions to the way RestEasy (classic) server extensions do it.

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml (hunk 1 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/pom.xml]: done

**Review by FroMage — APPROVED** [review]

**Comment by geoand** [comment]
Don't we want this in `1.13`?

The REST Client without this is kinda useless IMHO
